geomesa-datastore-bundle/geomesa-datastore-processors/src/main/scala/org/geomesa/nifi/datastore/processor/records/GeoAvroRecordSetWriterFactory.scala

```diff
@@ -0,0 +1,118 @@
+package org.geomesa.nifi.datastore.processor.records
+
+import java.io.OutputStream
+import java.util
+
+import org.apache.nifi.annotation.documentation.{CapabilityDescription, Tags}
+import org.apache.nifi.components.PropertyDescriptor
+import org.apache.nifi.controller.AbstractControllerService
+import org.apache.nifi.expression.ExpressionLanguageScope
+import org.apache.nifi.logging.ComponentLog
+import org.apache.nifi.processor.util.StandardValidators
+import org.apache.nifi.schema.access.SchemaNameAsAttribute
+import org.apache.nifi.serialization.record.{Record, RecordSchema}
+import org.apache.nifi.serialization.{AbstractRecordSetWriter, RecordSetWriterFactory}
+import org.geomesa.nifi.datastore.processor.records.GeoAvroRecordSetWriterFactory.{GeometryColumns, VisibilitiesColumn, TypeName}
+import org.geomesa.nifi.datastore.processor.records.GeometryEncoding.GeometryEncoding
+import org.geomesa.nifi.datastore.processor.records.SimpleFeatureRecordConverter.TypeAndEncoding
+import org.locationtech.geomesa.features.avro.AvroDataFileWriter
+
+import scala.collection.JavaConverters._
+
+@Tags(Array("avro", "geoavro", "result", "set", "recordset", "record", "writer", "serializer", "row"))
+@CapabilityDescription("Writes the contents of a RecordSet as GeoAvro which AvroToPutGeoMesa* Processors can use.")
+class GeoAvroRecordSetWriterFactory extends AbstractControllerService with RecordSetWriterFactory {
+  // NB: This is the same as what the InheritSchemaFromRecord Strategy does
+  override def getSchema(map: util.Map[String, String], recordSchema: RecordSchema): RecordSchema = recordSchema
+
+  override def createWriter(componentLog: ComponentLog,
+                            recordSchema: RecordSchema,
+                            outputStream: OutputStream,
+                            map: util.Map[String, String]): GeoAvroRecordSetWriter = {
+    new GeoAvroRecordSetWriter(componentLog, recordSchema, outputStream, getConfigurationContext.getProperties)
+  }
+
+  override def getSupportedPropertyDescriptors: util.List[PropertyDescriptor] =
+    Seq(GeometryColumns, VisibilitiesColumn, TypeName, RecordIngestProcessor.Properties.GeometrySerialization).toList.asJava
```

Comment:
@elahrvivaz thoughts on how to best use properties which are common to the RecordIngestProcessors and the GeoAvroRecordSetWriter?

In the last push, I am trying out sharing the GeometrySerialization one.  The VisibilitiesColumn one may be a good thing to duplicate as well.  (I unintentionally created a second one of these...)

Reply:
i'd probably put them in an `object Properties { ??? }` in the org.geomesa.nifi.datastore.processor.record package object